client/mysql.cc

```diff
@@ -1704,8 +1704,8 @@ static struct my_option my_long_options[] =
    &opt_local_infile, &opt_local_infile, 0, GET_BOOL, OPT_ARG, 0, 0, 0, 0, 0, 0},
   {"no-beep", 'b', "Turn off beep on error.", &opt_nobeep,
    &opt_nobeep, 0, GET_BOOL, NO_ARG, 0, 0, 0, 0, 0, 0},
-  {"host", 'h', "Connect to host.", &current_host,
-   &current_host, 0, GET_STR_ALLOC, REQUIRED_ARG, 0, 0, 0, 0, 0, 0},
+  {"host", 'h', "Connect to host. If host is not specified, connect to localhost by default or $MARIADB_HOST",
```

Comment:
I suspect it would read better if this said something along the lines of "Defaults in the following order: environment variables MARIADB_HOST, MYSQL_HOST, and then localhost."